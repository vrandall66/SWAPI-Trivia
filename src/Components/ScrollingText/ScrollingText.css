.body {
  display: flex;
  align-content: center;
  width: 100%;
  height: 20%;
  justify-content: center;
  overflow: hidden;
  margin-top: 43px;
}

.scrolling-Text {
  align-items: center;
  display: flex;
  justify-content: center;
  height: 150px;
  perspective: 200px;
  color: #FFE81F;
  font-family: "Pathway Gothic One", sans-serif;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 300%;
  text-align: justify;
  width: 100%;
}

.scrollingText--Episode-info {
  justify-content: center;
}

h1 {
  text-transform: uppercase;
}

.crawl {
  position: relative;
  text-transform: uppercase;
  transform-origin: 50% 100%;
  animation: crawl 400s linear;
  width: 250%;
}

@keyframes crawl {
  0% {
    top: 200px;
    transform: rotateX(20deg) translateZ(120px);
  }
  100% {
    top: -6000px;
    transform: rotateX(60deg) translateZ(-2500px);
  }
}
