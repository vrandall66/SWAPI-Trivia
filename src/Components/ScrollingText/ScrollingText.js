import React from "react";
import "./ScrollingText.css";

const ScrollingText = ({ episode }) => {
  return (
    <div className="body">
      <section className="scrolling-Text">
        <div className="crawl">
          <div className="scrollingText--Episode-info">
            <h1>
              Episode {episode.episode_id} {episode.title}
            </h1>
          </div>
          <h1>{episode.opening_crawl}</h1>
        </div>
      </section>
    </div>
  );
};

export default ScrollingText;
