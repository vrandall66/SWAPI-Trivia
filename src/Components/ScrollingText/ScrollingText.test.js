import React from "react";
import { shallow } from "enzyme";
import ScrollingText from "./ScrollingText";

describe("ScrollingText", () => {
  let wrapper;
  let episode = {
    characters: [],
    created: "",
    director: "",
    edited: "",
    episode_id: 0,
    opening_crawl: "",
    planets: [],
    producer: "",
    releaseDate: "",
    species: [],
    starships: [],
    title: "The Phantom Menace",
    url: "",
    vehicles: []
  };

  beforeEach(() => {
    wrapper = shallow(<ScrollingText episode={episode} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
