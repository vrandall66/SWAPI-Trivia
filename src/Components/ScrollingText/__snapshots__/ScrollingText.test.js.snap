// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should match the snapshot 1`] = `
<div
  className="body"
>
  <section
    className="scrolling-Text"
  >
    <div
      className="crawl"
    >
      <div
        className="scrollingText--Episode-info"
      >
        <h1>
          Episode 
          0
           
          The Phantom Menace
        </h1>
      </div>
      <h1 />
    </div>
  </section>
</div>
`;
